### PR TITLE
Join branch

### PR DESCRIPTION
**Pull request by soesandarwin2201**
## What does this PR do?
This PR is raised to add some new tables and add foreign key columns to existing animals table to model one-to-many relationships. Afterward will use JOIN to query the data.

## Description of the task to be completed
- [x] Create a table named owners
- [x] Create a table named species
- [x] Modify animals table
- [x] Write queries (using JOIN)

## How should it be tested
Clone the repository
```
git clone  https://github.com/soesandarwin2201/vet_clinic.git
```
Navigate to the folder
```
cd vet_clinic
```
Checkout the branch
```
git checkout join-branch

```
Screenshot for database 

Modify Screen shot

- [x]  What animals belong to Melody Pond?

Screenshot (348)

- [x] List of all animals that are pokemon (their type is Pokemon). 

Screenshot (350)

- [x] List all owners and their animals, remember to include those that don't own any animal. 
Screenshot (352)

- [x] How many animals are there per species? 
Screenshot (354)

- [x] List all Digimon owned by Jennifer Orwell.
Screenshot (356)

- [x] List all animals owned by Dean Winchester that haven't tried to escape.
Screenshot (358)

- [x] Who owns the most animals?
Screenshot (360)

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -36,3 +36,41 @@ SELECT neutered,MAX(escape_attempts) FROM animals GROUP BY neutered;
...
+
+UPDATE animals SET owner_id = 1 WHERE name = 'Agumon';
+UPDATE animals SET owner_id = 2 WHERE name IN ('Gabumon','Pikachu');
+UPDATE animals SET owner_id = 3 WHERE name IN ('Charmander','Squirtle','Blossom');
+UPDATE animals SET owner_id = 4 WHERE name IN ('Angemon','Boarmon');
+
+SELECT animals.name
+FROM animals
+JOIN owners ON owners.id = animals.owner_id 
+WHERE owners.full_name = 'Melody Pond';
```
Comment (Nemwel-Boniface) [inline queries.sql:50]: - [ ] The query provided and the one in the screenshot are two different queries. Kindly update either your screenshot or the query.

**This is the one in the screenshot:**

image

File: queries.sql [hunk 2]
```diff
@@ -36,3 +36,41 @@ SELECT neutered,MAX(escape_attempts) FROM animals GROUP BY neutered;
...
+SELECT animals.name
+FROM animals
+JOIN species ON species.id = animals.species_id 
+WHERE species.name = 'Pokemon';
+SELECT owners.full_name ,animals.name
+FROM animals
+RIGHT JOIN owners ON owners.id = animals.owner_id;
```
Comment (Nemwel-Boniface) [inline queries.sql]: - [x] When listing all the owners including the ones who do not have an animal, you are supposed to select from the owners table and then join the animals' table. Kindly update your query accordingly.

File: queries.sql [hunk 3]
```diff
@@ -36,3 +36,41 @@ SELECT neutered,MAX(escape_attempts) FROM animals GROUP BY neutered;
...
+GROUP BY (species.name);
+SELECT owners.full_name, animals.name, species.name
+FROM animals
+JOIN owners ON owners.id = animals.owner_id
+JOIN species ON animals.species_id = species.id
+WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
+SELECT owners.full_name, animals.name, animals.escape_attempts
+FROM animals
+JOIN owners ON owners.id = animals.owner_id
+WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0;
```
Comment (Nemwel-Boniface) [inline queries.sql:70]: - [ ] Kindly include the screenshot for this query as I have not seen in your pull request description. Or kindly label the screenshots properly as I have requested you to do in the pull request summary

image

File: queries.sql [hunk 4]
```diff
@@ -36,3 +36,41 @@ SELECT neutered,MAX(escape_attempts) FROM animals GROUP BY neutered;
...
+SELECT owners.full_name, COUNT(animals.name)
+FROM animals
+JOIN owners ON animals.owner_id = owners.id
+GROUP BY (owners.full_name) 
+ORDER BY count DESC
+LIMIT 1;
```
Comment (Nemwel-Boniface) [inline queries.sql:76]: - [ ] Kindly include the screenshot for this query as I have not seen in your pull request description. Or kindly label the screenshots properly as I have requested you to do in the pull request summary

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @soesandarwin2201        :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] You seem to have forgotten to insert the following data to your data.sql file. Kindly include them.

image

- [x] Kindly ensure that you label your screenshots having the query it is supposed to be for on top of the screenshot. Your provided screenshots are jumbled up and are hectic to look for the query in particular as they do not have a specific order as they are in the project requirements.




_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (soesandarwin2201) [reply queries.sql:76]: @Nemwel-Boniface Thanks for pointing out in Detail and give me the chance to improve my code . Thanks to your detail suggestion , I understand what did I wrong and learn that.Thanks for your time and effort . I am gonna change them right now

Comment (Nemwel-Boniface) [reply queries.sql:76]: good job :clap:

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @soesandarwin2201👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**